torch/optim/lr_scheduler.py

```diff
@@ -259,6 +263,11 @@ def __init__(self, optimizer, T_max, eta_min=0, last_epoch=-1):
     def get_lr(self):
         if self.last_epoch == 0:
             return self.base_lrs
+        elif (self.last_epoch - 1 - self.T_max) % (2 * self.T_max) == 0:
+            return [group['lr'] + (base_lr - self.eta_min) *
+                    (1 - math.cos(math.pi / self.T_max)) / 2
+                    for base_lr, group in
+                    zip(self.base_lrs, self.optimizer.param_groups)]
```

Comment:
I spent a while staring at this for a while and couldn't figure out if the equation was right. It must be right, since the tests pass. Would you mind saying a little more about the derivation here?

Reply:
![latex_4a12231ae15a2d9abc1c42fea250a362](https://user-images.githubusercontent.com/6797874/56165000-e0f37080-5f9f-11e9-8ff3-405eded9c15c.png)

Reply:
Awesome, clears it up!